packages/node/src/policy-gen/policy-gen-context.js

```diff
@@ -269,16 +315,24 @@ export class PolicyGeneratorContext {
    * @returns {LavamoatModuleRecord[]} Zero or more LMRs
    */
   buildModuleRecordsFromImportMap(importMap) {
-    return keys(importMap)
-      .filter((specifier) => this.isBuiltin(specifier))
-      .map((specifier) =>
-        this.#lmrCache.get({
+    return keys(importMap).reduce((acc, specifier) => {
+      if (this.isBuiltin(specifier)) {
+        /** @type {SimpleLavamoatModuleRecordOptions} */
+        const lmrOpts = {
           type: LMR_TYPE_BUILTIN,
           file: specifier,
           specifier,
-          packageName: getPackageName(this.compartment, this.#isEntry),
-        })
-      )
+          packageName: this.canonicalName,
```

Comment:
wait, if this is a builtin and we've found it in import map of the current context, the canonicalName is pointing to the package and is wrong for this entry.

Should it be hardcoded to 'node:' or null (to indicate it wants no top-level entry in resources for itself)?

Reply:
yes, you're right.  this field is seemingly ignored by `core` for builtins, so I guess it doesn't really matter.  according to the docs that presumably I wrote, it should be equal to `specifier`.  so I changed it to `specifier`.